server/src/main/java/org/opensearch/common/cache/tier/DiskTierTookTimePolicy.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.common.cache.tier;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.common.bytes.BytesReference;
+
+import java.util.function.Function;
+
+/**
+ * A cache tier policy which accepts queries whose took time is greater than some threshold,
+ * which is specified as a dynamic cluster-level setting. The threshold should be set to approximately
+ * the time it takes to get a result from the cache tier.
+ * The policy expects to be able to read a CachePolicyInfoWrapper from the start of the BytesReference.
+ */
+public class DiskTierTookTimePolicy implements CacheTierPolicy<BytesReference> {
+    public static final Setting<TimeValue> DISK_TOOKTIME_THRESHOLD_SETTING = Setting.positiveTimeSetting(
+        "indices.requests.cache.disk.tooktime.threshold",
+        TimeValue.ZERO,
```

Comment:
Do we need to do some benchmarking tests to better understand what should be the default value?

Reply:
Yes - this is one of the planned benchmarking scenarios for tiered caching. We plan to have a range of queries that take different lengths of time, and see how short that time can get before sending queries to the disk tier stops providing any benefit. The current value is a placeholder. The benchmarking scenarios are on hold right now because of changes being made to some of Sagar's tiered caching PRs.